controllers/componentdetectionquery_controller.go

```diff
@@ -320,11 +324,22 @@ func (r *ComponentDetectionQueryReconciler) Reconcile(ctx context.Context, req c
 
 				devfilesMapReturned, devfilesURLMapReturned, dockerfileContextMapReturned, componentPortsMapReturned, revision, err = cdqanalysis.CloneAndAnalyze(k8sInfoClient, req.Namespace, req.Name, context, cdqInfo)
 				if err != nil {
-					log.Error(err, fmt.Sprintf("Error running cdq analysis... %v", req.NamespacedName))
+					switch err.(type) {
+					case *cdqanalysis.NoDevfileFound:
+						log.Error(err, fmt.Sprintf("NoDevfileFound error running cdq analysis... %v", req.NamespacedName))
+					case *cdqanalysis.NoDockerfileFound:
+						log.Error(err, fmt.Sprintf("NoDockerfileFound error running cdq analysis... %v", req.NamespacedName))
+					default:
+						// Increment the git import failure metric only on non user failure
+						metrics.ImportGitRepoFailed.Inc()
+						log.Error(err, fmt.Sprintf("Internal error running cdq analysis... %v", req.NamespacedName))
+					}
+
 					r.SetCompleteConditionAndUpdateCR(ctx, req, &componentDetectionQuery, copiedCDQ, err)
 					return ctrl.Result{}, nil
 				}
 			}
+			metrics.ImportGitRepoSucceeded.Inc()
```

Comment:
As mentioned above, we should still be incrementing the success counter even in cases of user errors

Reply:
updated